src/apps/invitations/services.py

```diff
@@ -56,6 +58,40 @@ def __init__(self, session, user: User):
     async def fetch_all(self, query_params: QueryParams) -> list[InvitationDetail]:
         return await self.invitations_crud.get_pending_by_invitor_id(self._user.id, query_params)
 
+    async def fetch_by_emails(self, emails: list[str]) -> dict[str, InvitationDetail]:
+        return await self.invitations_crud.get_latest_by_emails(emails)
+
+    async def fill_pending_invitations_respondents(
+        self, respondents: list[WorkspaceRespondent]
+    ) -> list[WorkspaceRespondent]:
+        emails = [respondent.email for respondent in respondents if respondent.email is not None]
+        invitations = await self.fetch_by_emails(emails)
+
+        for respondent in respondents:
+            for detail in respondent.details or []:
+                if f"{respondent.email}_{detail.applet_id}" in invitations:
+                    invitation = invitations[
+                        "{email}_{applet_id}".format(email=respondent.email, applet_id=detail.applet_id)
+                    ]
+                    detail.__dict__["invitation"] = InvitationResponse(
```

Comment:
I avoid a direct assignment here because `WorkspaceRespondentDetails` has `@root_validator`s that do decryption of some fields.

https://github.com/ChildMindInstitute/mindlogger-backend-refactor/blob/16db3ce7d294a621b27547694ecc3d4334c34a30/src/apps/workspaces/domain/workspace.py#L83-L92

But the fields I'm assigning are already decrypted, so the validators fail. Do you have a suggestion of another way I might work around that?

Reply:
`InvitationResponse` is a PublicModel and `WorkspaceRespondentDetails` is a InternalModel, you need to create a new InternalModel with just what you need from invitation

Reply:
I don't understand what you mean. The issue is with `WorkspaceRespondentDetails`, not `InvitationResponse`. It's the one that uses the root validators (that are unrelated to the invitation). How would creating a new internal model help with that?

Reply:
Just to chime in here: if it's agreed that `detail.__dict__["invitation"] = …` assignment makes sense in the end, I think it warrants a comment to explain why it's done that way. But I'll leave you to discuss. 😉

Reply:
Please, do not merge with it, this code is mixing InternalModel with PublicModel, I will investigate a solution for this tomorrow and post here